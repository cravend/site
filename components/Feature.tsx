import React, { FC, ReactChild } from 'react';
import { Card, Heading, Text } from 'theme-ui';

type FeatureType = {
  heading: string,
  body: ReactChild,
}

const Feature:FC<FeatureType> = ({ heading, body }: FeatureType) => (
  <Card variant="interactive">
    <Heading as="h2" mb={[1, 2]}>{heading}</Heading>
    <Text>{body}</Text>
  </Card>
);

export default Feature;
