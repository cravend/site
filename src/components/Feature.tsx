import styles from "../styles/Feature.module.scss";
import type { FC, ReactNode } from "react";

type FeatureType = {
  heading: string;
  children: ReactNode;
};

const Feature: FC<FeatureType> = ({ heading, children }: FeatureType) => (
  <div className={styles.card}>
    <h2 className={styles.heading}>{heading}</h2>
    {children}
  </div>
);

export default Feature;
