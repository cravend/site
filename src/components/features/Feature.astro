---
export interface Props {
  header: string;
}

const { header } = Astro.props;
---

<!-- TODO: should this be <article>? -->
<section>
  <h3>{header}</h3>
  <slot />
</section>
