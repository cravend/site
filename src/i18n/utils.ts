import { locales, defaultLocale, RTL_LOCALES } from "./config";

import type { Locale } from "./config";
import type { NextRouter } from "next/router";

/**
 * @description This function is used to check if a string is a valid locale.
 *
 * @param localeToTest The string to check.
 * @returns True if the string is a valid locale, false otherwise.
 */
export const isLocale = (localeToTest?: string): localeToTest is Locale =>
  locales.some((locale) => locale === localeToTest);

/**
 * @description This function is used to get the current locale.
 *
 * If a custom locale is set by \@next/router, it will be returned.
 * If no custom locale is set, the router's default locale will be returned.
 * If the router's default locale isn't set, the default locale will be returned.
 *
 * @param router The NextRouter instance.
 * @returns The current locale.
 */
export const getLocale = (router: NextRouter | null) => {
  if (!router) return defaultLocale;
  if (isLocale(router.locale)) return router.locale;
  if (isLocale(router.defaultLocale)) return router.defaultLocale;
  return defaultLocale;
};

/**
 * @description This function is used to toggle the current locale.
 *
 * @param locale The current locale to be toggled.
 * @returns The new locale.
 */
export const toggleLocale = (locale: Locale) =>
  locale === locales[0] ? locales[1] : locales[0];

/**
 * @description This function is used to get the text direction of the current locale.
 *
 * @param locale The locale to be checked
 * @returns True if the locale is read right-to-left, false otherwise
 */
export const isRtl = (locale: Locale) => RTL_LOCALES.includes(locale);
