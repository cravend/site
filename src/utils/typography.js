import Typography from 'typography'

const typography = new Typography({
  baseLineHeight: 1.666,
  headerFontFamily: ['Poppins', 'Helvetica', 'Arial', 'sans-serif'],
  bodyFontFamily: ['Poppins', 'Helvetica', 'Arial', 'sans-serif'],
  googleFonts: [
    {
      name: 'Poppins',
      styles: ['300', '600', '700']
    }
  ]
})

export default typography
