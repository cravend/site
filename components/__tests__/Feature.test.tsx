import React from "react";
import { screen, render } from "../../src/tests/utils";
import Feature from "../Feature";

describe("<Feature />", () => {
  it("renders", () => {
    expect.assertions(1);
    const FEATURE_HEADING = "Feature heading";
    const CHILD_TEXT = "Hello, world!";

    const { container } = render(
      <Feature heading={FEATURE_HEADING}>
        <p>{CHILD_TEXT}</p>
      </Feature>
    );
    expect(container).toMatchSnapshot();
  });

  it("displays text", () => {
    expect.assertions(1);
    const FEATURE_HEADING = "Feature heading";
    const CHILD_TEXT = "Hello, world!";

    render(
      <Feature heading={FEATURE_HEADING}>
        <p>{CHILD_TEXT}</p>
      </Feature>
    );

    expect(
      screen.getByRole("heading", { name: FEATURE_HEADING })
    ).toBeInTheDocument();
  });

  it("displays a level-two heading", () => {
    expect.assertions(1);
    const FEATURE_HEADING = "Feature heading";
    const CHILD_TEXT = "Hello, world!";

    render(
      <Feature heading={FEATURE_HEADING}>
        <p>{CHILD_TEXT}</p>
      </Feature>
    );

    expect(
      screen.getByRole("heading", { name: FEATURE_HEADING, level: 2 })
    ).toBeInTheDocument();
  });

  it("displays child components", () => {
    expect.assertions(1);
    const FEATURE_HEADING = "Feature heading";
    const CHILD_TEXT = "Hello, world!";

    render(
      <Feature heading={FEATURE_HEADING}>
        <p>{CHILD_TEXT}</p>
      </Feature>
    );

    expect(screen.getByText(CHILD_TEXT)).toBeInTheDocument();
  });
});
