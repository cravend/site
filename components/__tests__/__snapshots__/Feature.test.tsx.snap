// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feature /> renders 1`] = `
<div>
  <div
    class="css-vurnku"
  >
    <h2
      class="css-qyfm59"
    >
      Feature heading
    </h2>
    <p>
      Hello, world!
    </p>
  </div>
</div>
`;
